src/upstream.js

```diff
@@ -16,14 +16,32 @@ export class Upstream {
 		this[_proxy] = opts.proxy;
 	}
 
-	handle (req, res) {
-		this[_proxy].web(req, res, {
-			target: url.format({
-				protocol: this.protocol,
-				hostname: this.hostname,
-				port: this.port,
-				path: this.path
-			})
-		});
+	handle () {
+		var req = arguments[0];
+		// checks the number of arguments to see if this is a standard request or a websocket request
+		if (arguments.length === 3) {
```

Comment:
Not a fan of this.  Arity based checks are a bad pattern, express core is actually going away from this because of the issues it causes.  I think it needs explicit methods for handling the different types of requests.